server/reload.go

```diff
@@ -115,41 +233,86 @@ func (s *Server) diffOptions(newOpts *Options) ([]option, error) {
 		}
 		switch strings.ToLower(field.Name) {
```

Comment:
@tylertreat , with the refactoring there could be if statements calling a convenience API (e.g. `isUserOption(string)`, and then switch statements within each function.  A tradeoff between code purity and a very large function.  It's not something necessary now, and shouldn't hold up this PR, but I would guess that this function will need to be refactored as your reload work handles more configuration options.